### PR TITLE
Tickets/dm 33674

### PR DESCRIPTION
**Pull request by pcortes80**
- Create the `communicationDiagramQueues.md` file.
- Create images and `.uml` files to support the `communicationDiagramQueues.md` file.
- Please note that the `userManual.md` file is modified. However this change is not part of this ticket, because I replaced an image by mistake.

### CODE REVIEW

**Review by teweitsai — APPROVED** [review]
This document looks good. Please reformat the `communicationDiagramQueues.md` by using the lists. In addition, try to avoid using too many words of "the".

I think we have a ticket to test the adding of diagrams programmatically. I would like you use these document to do the related tests because I do not think it is a good idea to hold the png files in the repository. They will be out-of-sync quickly.

Thanks!

File: doc/uml/commandToComponent.uml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+rectangle queueTcpServer as QTS #F694CE
+
+TC --> TSC : g1. write_cmd_and_wait_result()\nTCP Packet
+TSC --> QTS : g2. tcpServerStateMachine()\nqueueTcpServer
+QTS --> TSC : g3. tcpServerCallBack()\nTCP Packet
+TSC --> TSC : g4. processPacket()\nTCP Packet
+TSC --> Component : g5. processCmd()\nCmdRead UserEvent
+
+@enduml
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/commandToComponent.uml:16]: Done.

File: doc/uml/getQueueStatus.uml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+QTC --> TST : d1. GetQueueStatus()\nqueueTcpClient
+QTCFD --> TST : e1. GetQueueStatus()\nqueueTcpClientFromDds
+TST --> TS : d2. checkQueueFull()\nqueueTcpClient
+TST --> TS : e2. checkQueueFull()\nqueueTcpClientFromDds
+
+TS --> Component : d3. isQueueTcpClientFull()\nqueueFull
+TS --> Component : e3. isQueueTcpClientFull()\nqueueFull
+
+@enduml
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/getQueueStatus.uml:19]: Done.

File: doc/uml/getTelElement.uml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+rectangle TcpServer as TS
+rectangle TcpServerTel as TST
+
+rectangle queueTcpClientFromDds as QTCFD #b0e0e6
+
+QTCFD --> TST : f1. DequeueElement()\nqueueTcpClientFromDds
+TST --> TS : f2. getTel()\nqueueClientFromDds
+TS --> Component : f3. getTelElement()\nqueueTcpClientFromDds
+
+@enduml
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/getTelElement.uml:14]: Done.

[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/getTelElement.uml:14]: Done before.

File: doc/uml/telemetryToClient.uml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+rectangle TcpServerBase as TSB
+
+rectangle queueTcpClient as QTC #ffba71
+
+Component --> TS : a1. sendTel()\ntelTypeDef
+TS --> TST : a2. telToClient()\nJSON string
+TST --> QTC : a3. EnqueueElement()\nqueueTcpClient
+QTC --> TSB : a4. DequeueElement()\nqueueTcpClient
+TSB --> TC : a5. Write TCP()\nJSON string
+@enduml
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/telemetryToClient.uml:17]: Done.

File: doc/uml/telemetryToComponent.uml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+rectangle queueTcpClientFromDds as QTCFD #b0e0e6
+
+TC --> TST : h1. tcp_client_tel.write()\nTCP Packet
+TST --> QTS : h2. tcpServerStateMachine()\nqueueTcpServer
+QTS --> TST : h3. processTel()\nTCP Packet
+TST --> QTCFD : h4. enqueueTelToComponent()\nqueueTcpClientFromDds
+QTCFD --> TS : h5. getTel()\nTCP Packet
+TS --> Component : h6. getTelElement()\nTCP Packet
+
+@enduml
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/uml/telemetryToComponent.uml:19]: Done.

File: doc/communicationDiagramQueues.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+# Queues Communication Diagram
+
+## Overview
+
+This Communication Diagram shows how the **TcpServer** uses queues internally to share data among classes.
+
+## Diagram Conventions
+
+The [PlantUML](https://plantuml.com) is used to draw the Communication Diagram.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:9]: I noticed that PlantText is not too stable. I prefer to recommend PlantUML only. Thanks.

File: doc/communicationDiagramQueues.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+| Queue Name |
+|:----------:|
+| queueTcpServer    |
+| queueTcpClient    |
+| queueTcpClientFromDds    |
+
+## Queues Description
+
+The following section will describe the queues:
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:24]: Done.

File: doc/communicationDiagramQueues.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+- The `queueTcpServer` is a **TcpServerBase** attribute. It stores commands, events and telemetry comming from the TCP Client as TCP Packets.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:26]: Done.

File: doc/communicationDiagramQueues.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+- The `queueTcpClient` is a **TcpServerBase** attribute.
+It stores telemetry that is sent from the Component to the TCP Client.
+This queue has a max queue size set by the **TcpServer.configServer&#46;vi** with the `maxSizeQueueTcpClient` control.
+User can use the **queueOptionsTypeDef&#46;ctl** to set the Timeout (ms), and how to add new elements into this queue.
+- The `queueTcpClientFromDds` is a **TcpServerTel** attribute.
+It is used to receive the processed telemetry ready to be sent to the Component.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:32]: Done.

File: doc/communicationDiagramQueues.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+The **TcpServer.configServer&#46;vi** sets the max queue size with the `maxSizeQueueTcpClientFromDds` control, the timeout (ms) with the `timeoutQueueTcpClientFromDds` control, and how to add new elements into this queue, with the `EnqueueOptions` control.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:33]: Done.

File: doc/communicationDiagramQueues.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+| Queue | Enqueue | Dequeue |
+|:--------:|:----------:|:-----------:|
+| queueTcpServer  | TcpServerCmd.tcpServerStateMachine&#46;vi   | TcpServerCmd.tcpServerCallBack&#46;vi     |
+| queueTcpServer  | TcpServerTel.tcpServerStateMachine&#46;vi   | TcpServerTel.tcpServerCallBack&#46;vi     |
+| queueTcpClient  |  TcpServerTel.enqueueTelToClient&#46;vi  | TcpServerBase.tcpServerSenderLoop&#46;vi     |
+| queueTcpClientFromDds  | TcpServerTel.enqueueTelToComponent&#46;vi   | TcpServerTel.getTel&#46;vi     |
+
+## Communication Diagrams
+
+The following sections will present differents routs where queues interacts with classes.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:46]: Done.

File: doc/communicationDiagramQueues.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+| queueTcpClient    |
+| queueTcpClientFromDds    |
+
+## Queues Description
+
+The following section will describe the queues:
+
+- The `queueTcpServer` is a **TcpServerBase** attribute. It stores commands, events and telemetry comming from the TCP Client as TCP Packets.
+- The `queueTcpClient` is a **TcpServerBase** attribute.
+It stores telemetry that is sent from the Component to the TCP Client.
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:28]: Done.

File: doc/communicationDiagramQueues.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+
+## Communication Diagrams
+
+The following sections will present differents routs where queues interacts with classes.
+
+### Send Telemetry from Component to TCP Client
+
+User sends the telemetry to the TCP Client with the **sendTel()&#46;vi** (a1) using the **telTypeDef&#46;ctl**.
+The **telToClient&#46;vi** (a2) converts the telemetry into a JSON string, and then the **enqueueTelToClient&#46;vi** (a3) put the telemetry into the `queueTcpClient` queue.
+In (a4) the **TcpServerBase.tcpServerSenderLoop&#46;vi** instance dequeues the telemetry as a JSON string and is ready to be sent to the TCP Client with the **Write TCP** LabVIEW function (a5).
```
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/communicationDiagramQueues.md:52]: Got it. Thanks.

**Pull request merged**